Servers/Administration/OJS.Servers.Administration/Controllers/ProblemResourcesController.cs

```diff
@@ -0,0 +1,186 @@
+namespace OJS.Servers.Administration.Controllers;
+
+using AutoCrudAdmin.Extensions;
+using AutoCrudAdmin.Models;
+using AutoCrudAdmin.ViewModels;
+using FluentExtensions.Extensions;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using OJS.Data.Models.Problems;
+using OJS.Servers.Infrastructure.Extensions;
+using OJS.Services.Administration.Business.Extensions;
+using OJS.Services.Administration.Business.Validation;
+using OJS.Services.Administration.Business.Validation.Factories;
+using OJS.Services.Administration.Data;
+using OJS.Services.Administration.Models;
+using OJS.Services.Administration.Models.ProblemResources;
+using OJS.Services.Common;
+using OJS.Services.Infrastructure.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Threading.Tasks;
+
+public class ProblemResourcesController : BaseAutoCrudAdminController<ProblemResource>
+{
+    private const string ProblemIdKey = nameof(ProblemResource.ProblemId);
+
+    private readonly IProblemResourceValidatorsFactory problemResourceValidatorsFactory;
+    private readonly IProblemResourcesDataService problemResourcesData;
+    private readonly IProblemResourcesDownloadValidationService problemResourcesDownloadValidation;
+    private readonly IContentTypesService contentTypes;
+
+    public ProblemResourcesController(
+        IProblemResourceValidatorsFactory problemResourceValidatorsFactory,
+        IProblemResourcesDataService problemResourcesData,
+        IProblemResourcesDownloadValidationService problemResourcesDownloadValidation,
+        IContentTypesService contentTypes)
+    {
+        this.problemResourceValidatorsFactory = problemResourceValidatorsFactory;
+        this.problemResourcesData = problemResourcesData;
+        this.problemResourcesDownloadValidation = problemResourcesDownloadValidation;
+        this.contentTypes = contentTypes;
+    }
+
+    public override IActionResult Index()
+    {
+        if (!this.TryGetEntityIdForColumnFilter(ProblemIdKey, out var problemId))
+        {
+            return base.Index();
+        }
+
+        var routeValues = new Dictionary<string, string>
+        {
+            { nameof(problemId), problemId.ToString() },
+        };
+
+        this.MasterGridFilter = pr => pr.ProblemId == problemId;
+        this.CustomToolbarActions = new AutoCrudAdminGridToolbarActionViewModel[]
```

Comment:
This is bad. Let's not do this. Setting instance properties inside a `Index` action is **hidden bahavior**.
This should be done declaratively, i.e.

```
// AutoCrudAdminController
protected ... MasterGridFilter = "default behavior"
protected ... CustomToolbarActions = "default behavior"

// ProblemResourcesController

protected override MasterGridFilter = pr => pr.ProblemId = ...
```

Reply:
How to get the `problemId` like this?

Reply:
For example, it may be a collection, add just append a handler here. Much like `formControls`.